Bookmark-Manager/src/test/java/com/bookmark/bookmarks/BookmarkServiceTest.java

```diff
@@ -0,0 +1,215 @@
+        bookmark.setUserId(UUID.randomUUID());
+
+        Bookmark updatedBookmark = new Bookmark();
+        updatedBookmark.setId(bookmarkId);
+        updatedBookmark.setTitle("Updated Title");
+        updatedBookmark.setUrl("http://example.com");
+        updatedBookmark.setUserId(UUID.randomUUID());
+
+        when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.of(bookmark));
+        when(bookmarkRepository.save(any(Bookmark.class))).thenReturn(updatedBookmark);
+
+        Bookmark savedBookmark = bookmarkService.updateBookmark(bookmarkId, updatedBookmark);
+
+        assertNotNull(savedBookmark);
+        assertEquals(bookmarkId, savedBookmark.getId());
+        assertEquals("Updated Title", savedBookmark.getTitle());
+        assertEquals("http://example.com", savedBookmark.getUrl());
+    }
+
+    @Test
+    void test_update_bookmark_with_invalid_id() {
+        UUID bookmarkId = UUID.randomUUID();
+        Bookmark updatedBookmark = new Bookmark();
+        updatedBookmark.setId(bookmarkId);
+        updatedBookmark.setTitle("Updated Title");
+        updatedBookmark.setUrl("http://example.com");
+        updatedBookmark.setUserId(UUID.randomUUID());
+
+        when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.empty());
+
+        try {
+            bookmarkService.updateBookmark(bookmarkId, updatedBookmark);
+        } catch (IllegalArgumentException e) {
+            assertEquals("Bookmark not found with ID: " + bookmarkId, e.getMessage());
+        }
+    }
+
+    @Test
+    void test_update_bookmark_with_invalid_data() {
+        UUID bookmarkId = UUID.randomUUID();
+        Bookmark bookmark = new Bookmark();
+        bookmark.setId(bookmarkId);
+        bookmark.setTitle("Sample Title");
+        bookmark.setUrl("http://example.com");
+        bookmark.setUserId(UUID.randomUUID());
+
+        Bookmark updatedBookmark = new Bookmark();
+        updatedBookmark.setId(bookmarkId);
+        updatedBookmark.setTitle("");
+        updatedBookmark.setUrl("http://example.com");
+        updatedBookmark.setUserId(UUID.randomUUID());
+
+        when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.of(bookmark));
+
+        try {
+            bookmarkService.updateBookmark(bookmarkId, updatedBookmark);
+        } catch (BookmarkManagerException e) {
+            assertEquals("Invalid Arguments", e.getMessage());
+        }
+    }
+
+    @Test
+    void test_delete_bookmark() {
+        UUID bookmarkId = UUID.randomUUID();
+        Bookmark bookmark = new Bookmark();
+        bookmark.setId(bookmarkId);
+        bookmark.setTitle("Sample Title");
+        bookmark.setUrl("http://example.com");
+        bookmark.setUserId(UUID.randomUUID());
+
+        when(bookmarkRepository.existsById(bookmarkId)).thenReturn(Boolean.TRUE);
+
+        Mockito.doNothing().when(bookmarkRepository).deleteById(bookmarkId);
+
+        bookmarkService.deleteBookmark(bookmarkId);
+
+
```

Comment:
Verification of Mock Interactions
Missing verification of mock interactions in some tests.

```
    verify(bookmarkRepository).deleteById(bookmarkId);
    verify(bookmarkRepository).existsById(bookmarkId);